Source/Things/Speed.as

```diff
@@ -75,6 +75,23 @@ class DashboardSpeed : DashboardThing
 				nvg::TextBox(Setting_Speed_Padding + textSize, m_size.y / 2, textSize, Icons::AngleDoubleRight + " " + Text::Format("%.0f", sideSpeed));
 				break;
 			}
+			
+			case SpeedStyle::Directional: {
+				float sideSpeed = Vehicle::GetSideSpeed(vis) * 3.6f;
+				float absSpeed = Math::Sqrt(Math::Pow(sideSpeed, 2) + Math::Pow(vis.FrontSpeed * 3.6f, 2));
+
+				float textSize = (m_size.x - Setting_Speed_Padding * 2) / 2;
+				nvg::TextAlign(nvg::Align::Middle | nvg::Align::Left);
+				nvg::TextBox(Setting_Speed_Padding, m_size.y / 2, textSize, Icons::ArrowUp + " " + Text::Format("%.0f", absSpeed));
+
+				float carAngle = Math::Abs((Math::Asin(vis.FrontSpeed * 3.6f / absSpeed) * 57) - 90) ;
```

Comment:
This throws a divide by zero exception when `absSpeed` is 0!

There's also a stray whitespace before the semicolon.